libraroo-backend/src/users/users.repository.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { User } from './models/user.model';
+import { Model } from 'mongoose';
+
+@Injectable()
+export class UsersRepository {
+  constructor(@InjectModel(User.name) private userModel: Model<User>) {}
+
+  async create(user: Partial<User>): Promise<User> {
+    const newUser = new this.userModel(user);
+    return newUser.save();
+  }
+
+  async findByEmail(email: string): Promise<User | null> {
+    return this.userModel.findOne({ email }).exec();
+  }
+}
```

Comment:
Lets try to improve the architecture, instead of duplicating alot of the repositories, implement a generic repo with all of the basic CRUD operations which will be inherited here, and only implement the specific ones like `findByEmail` in the `UsersRepository`.
Maybe try this example: https://chatgpt.com/share/6751cda6-5a58-800d-9e7f-fc9a246788cf (ctrl+f "Can you make the user repository generic?")